pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainScheduler.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.runtime.executor;
+
+import org.apache.pinot.query.mailbox.MailboxIdentifier;
+import org.apache.pinot.query.runtime.operator.OpChain;
+
+
+/**
+ * An interface that defines different scheduling strategies to work with the
+ * {@link OpChainSchedulerService}. All methods are thread safe and can be guaranteed
+ * to never be called concurrently - therefore all implementations may use data
+ * structures that are not concurrent.
+ */
+public interface OpChainScheduler {
+
+  /**
+   * @param operatorChain the operator chain to register
+   */
+  void register(OpChain operatorChain);
+
+  /**
+   * This method is called whenever {@code mailbox} has new data available to consume,
+   * this can be useful for advanced scheduling algorithms
+   *
+   * @param mailbox the mailbox ID
+   */
+  void onDataAvailable(MailboxIdentifier mailbox);
```

Comment:
is this the only trigger? (other than register)

Reply:
there's three possible triggers:
- register
- onDataAvailable
- next().getRoot().nextBlock() completes

Triggers are defined in the implementation instead of the interface.